lib/fs/debug.go

```diff
@@ -19,11 +16,7 @@ var (
 
 func init() {
 	logger.DefaultLogger.NewFacility("walkfs", "Filesystem access while walking")
-	switch {
-	case strings.Contains(os.Getenv("STTRACE"), "walkfs") || os.Getenv("STTRACE") == "all":
-		l.SetDebug("walkfs", true)
-		fallthrough
-	case strings.Contains(os.Getenv("STTRACE"), "fs"):
+	if logger.DefaultLogger.ShouldDebug("walkfs") {
```

Comment:
Just setdebug(... ShouldDebug())

Reply:
Also, do we need this? Because "fs" is in "walkfs"

Reply:
`l.SetDebug("fs", logger.DefaultLogger.ShouldDebug("walkfs"))` would disable `fs` tracing if `walkfs` is not traced.

Relying on "fs" being in "walkfs" feels like a hack. Also, [postSystemDebug](https://github.com/syncthing/syncthing/blob/master/lib/api/api.go#L615) can set just `walkfs` without setting `fs`. Sadly, this code is run just once before postSystemDebug.